app/pulp/app/models/progress.py

```diff
@@ -18,6 +20,16 @@ class ProgressReport(Model):
 
     All progress reports have a message, state, and are related to a Task.
 
+    Attributes:
+
+        batch_increments (bool): If this Flag is set save checks to make sure it only gets ran once
```

Comment:
I think we should allow plugin writer the ability to control whether or not they want to batch their save() calls. I did update the docs to call out the fact that it is up to a plugin writer has to ensure that the last save() is called. What do you think about this?